File: 3. lab/3.lab.14/src/main/java/plugins/Parrot.java
package plugins;

import model.Animal;

/**
 * @author Antonio Bukovac
 */
public class Parrot extends Animal {

    private String name;

    public Parrot(String name) {
        this.name = name;
    }

    @Override
    public String name() {
        return name;
    }

    @Override
    public String greet() {
        return "Bok ja sam papiga";
    }

    @Override
    public String menu() {
        return "Volim sjemenke suncokreta";
    }

}

File: 3. lab/1.1/parrot.c
#include <stdlib.h>
#include<malloc.h>
#include <stdio.h>
typedef char const* (*PTRFUN)();

typedef struct {
	PTRFUN* vtable;
	char const* name;
} Parrot;

char const* name(Parrot* parrot){
    return parrot->name;
}

char const* greet(void){
    return "Bok ja sam papiga!";
}

char const* menu(void){
    return "Sjemenke suncokreta";
}

PTRFUN vtable[] = {name, greet, menu};

void construct(Parrot* parrot, char const* name){
    parrot->vtable = vtable;
    parrot->name = name;
}

void* create(char const* name){
    Parrot* parrot = (Parrot*) malloc(sizeof(Parrot));
    construct(parrot, name);
    return parrot;
}
